src/main/java/spoon/support/adaption/TypeAdaptor.java

```diff
@@ -150,11 +195,51 @@ public static boolean isSubtype(CtType<?> base, CtTypeReference<?> superRef) {
 		if (useLegacyTypeAdaption(base)) {
 			return new TypeAdaptor(base).isSubtypeOf(superRef);
 		}
+
+		// Handle shadow array types, as we can build a CtType for any Class, which includes arrays. We need to lower
+		// them to their innermost component type and then decide subtyping based on their relationship.
+		if (base.isArray() && superRef instanceof CtArrayTypeReference<?>) {
+			if (!base.isShadow()) {
+				throw new SpoonException("There are no source level array type declarations");
+			}
+			Class<?> actualClass = base.getActualClass();
+			// Arrays of different shapes are not subtypes
```

Comment:
This isn't true?
> new Object[0][0] instanceof Object[]
$3 ==> true

This is a consequence of 
- If S and T are both reference types, then S[] >1 T[] iff S >1 T.
- Object >1 Object[]

unless I'm missing some context here.

Reply:
Addressed (hopefully :P). Thanks :)